cmd/build/v2/environment.go

```diff
@@ -23,17 +23,37 @@ import (
 )
 
 // SetServiceEnvVars set okteto build env vars
-func (*OktetoBuilder) SetServiceEnvVars(service, reference string) {
+func (bc *OktetoBuilder) SetServiceEnvVars(service, reference string) {
 	reg, repo, tag, image := registry.GetReferecenceEnvs(reference)
 
 	oktetoLog.Debugf("envs registry=%s repository=%s image=%s tag=%s", reg, repo, image, tag)
 
 	// Can't add env vars with -
 	sanitizedSvc := strings.ToUpper(strings.ReplaceAll(service, "-", "_"))
-	os.Setenv(fmt.Sprintf("OKTETO_BUILD_%s_REGISTRY", sanitizedSvc), reg)
-	os.Setenv(fmt.Sprintf("OKTETO_BUILD_%s_REPOSITORY", sanitizedSvc), repo)
-	os.Setenv(fmt.Sprintf("OKTETO_BUILD_%s_IMAGE", sanitizedSvc), reference)
-	os.Setenv(fmt.Sprintf("OKTETO_BUILD_%s_TAG", sanitizedSvc), tag)
+
+	registryKey := fmt.Sprintf("OKTETO_BUILD_%s_REGISTRY", sanitizedSvc)
+	bc.lock.Lock()
```

Comment:
I guess this lock prevents writes at the same time in `buildEnvironmnets` field but should we allow to rewrite same key again? maybe a lock before check if value is already set? or this lock has another purpose?

Reply:
The CLI will only set the name of the variables once, and We need to overwrite always the key because if not the user could set the variable on its terminal and we will be using the one that the user provided instead of the correct one.